src/player.cpp

```diff
@@ -1508,26 +1508,25 @@ void Player::checkTradeState(const Item* item)
 	}
 }
 
-void Player::setNextWalkActionTask(SchedulerTask* task)
+void Player::setNextWalkActionTask(SchedulerTask_ptr task)
 {
 	if (walkTaskEvent != 0) {
 		g_scheduler.stopEvent(walkTaskEvent);
 		walkTaskEvent = 0;
 	}
 
-	delete walkTask;
-	walkTask = task;
+	walkTask = std::move(task);
```

Comment:
I believe if you use

```suggestion
	std::swap(walkTask, task);
```

You may skip one deletion. This is absolutely minor, change it only if you touch this code again 😆

Reply:
why? deletion will be moved to end of the scope

Reply:
The swap looks strange.
Looking at it, I have no idea what author meant by doing swap... for me, it just obscures the code